webrtc/src/rtp_transceiver/mod.rs

```diff
@@ -424,11 +424,11 @@ impl RTCRtpTransceiver {
         self.stopped.store(true, Ordering::SeqCst);
 
         {
-            let sender = self.sender.lock();
+            let sender = self.sender.lock().clone();
```

Comment:
I think the usage of SyncMutex for self.sender and self.receiver is wrong here.

SyncMutex must not be used in await.

Could change self.sender and self.receiver from SyncMutex to tokio::sync::Mutex?

Reply:
something like
```
   sender: Arc<tokio::sync::Mutex<RTCRtpSender>>,     //atomic.Value
    receiver: Arc<tokio::sync::Mutex<RTCRtpReceiver>>, //atomic.Value
```

Reply:
I tried this and it works even without `clone()` https://github.com/webrtc-rs/webrtc/pull/429/commits/ef0f6c5f332e094e7ac4a2270e23086cce2f13cf

Reply:
> @yngrtc ,
> 
> > Could you try to change self.sender and self.receiver from SyncMutex to tokio::sync::Mutex?
> 
> This would revert #402 and introduce unnecessary async in public API.
> 
> > SyncMutex must not be used in across await
> 
> Mutex guard won't be held across await points if the `Arc` behind mutex is cloned. Its:
> 
> ```rust
> let sender: MutexGuard<_> = self.sender.lock().unwrap();
> let sender: Arc<_> = sender.clone();
> ```
> 
> Perhaps its better to explicitly use `Arc::clone(&*self.sender.lock().unwrap())` so it would be more clear that `stop()` is called on an `Arc` and not on a `MutexGuard`

Change it https://github.com/webrtc-rs/webrtc/pull/429/commits/b15f1cb4aa3408755c352137dd0740a9ba0d9831